### PR TITLE
Create adr-011-snapshot-tests.md

### PR DESCRIPTION
**Pull request by joshblack**
[Rendered markdown](https://github.com/primer/react/blob/8628606f1f9f5aaf419d33a998777a86f982e242/contributor-docs/adrs/adr-XXX-snapshot-tests.md)

**Links & Resources**

- https://kentcdodds.com/blog/effective-snapshot-testing
- https://github.com/jest-community/eslint-plugin-jest
  - https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-large-snapshots.md

### CODE REVIEW

**Comment by joshblack** [comment]
Feedback from @broccolinisoup 

- Check-in with others to see if there is a consensus on this
- Make sure to include section on impact of this (as there are parts of code that would need to change)
  - e.g. `behavesAsComponent` from `testing.tsx`
  - Broader point, are these utilities still helpful?

File: contributor-docs/adrs/adr-XXX-snapshot-tests.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+- Components with large or deep trees emit large snapshots
+- Debugging what change lead to a broken snapshots can be difficult
+- It is unclear what the intent may be of a snapshot test when something does fail, in other words it does not always answer the question: what is this testing?
+
+## Decision
+
+- Avoid using "catch-all" snapshot tests for React components
+- Prefer explicit snapshots, in particular for:
+  - A specific part of a component
```
Comment (broccolinisoup) [inline contributor-docs/adrs/adr-XXX-snapshot-tests.md:52]: Oh, this is cool! How can I snapshot a specific part of a component? 👀

Comment (joshblack) [reply contributor-docs/adrs/adr-XXX-snapshot-tests.md:52]: @broccolinisoup great question, I wish I could come up with more compelling use-cases lol. What comes to mind are generally things like class names or specific attributes in the rendered HTML.

**Review by colebemis — APPROVED** [review]
Love it. Can't wait to remove most of our snapshot tests. Might be nice to include an "Impact" section in this ADR explaining what we'll do with existing snapshot tests if we accept this decision.

**Comment by joshblack** [comment]
Thanks @colebemis! Just updated with a small impact section, let me know if it'd be helpful to go into more detail

**Comment by colebemis** [comment]
@joshblack Looks good!

**Review by broccolinisoup — APPROVED** [review]
This looks looking real good ✨

File: contributor-docs/adrs/adr-XXX-snapshot-tests.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+</td></tr>
+</tbody></table>
+
+### Impact
+
+This decision will impact our test suite in two ways:
+
+- Tests with calls to `toMatchSnapshot()` may need to be removed if the snapshot
+  test is capturing the render tree
+- The `behavesAsComponent` helper, which uses `toMatchSnapshot` under-the-hood,
```
Comment (broccolinisoup) [inline contributor-docs/adrs/adr-XXX-snapshot-tests.md:84]: This was one of the action items from the eng sync that I plan to follow up. Should we create an issue for the impact or is it not necessary?

**Comment by joshblack** [comment]
**Feedback**

- Explicitly call out that this is a rare situation
- Try and provide more examples 
- Explicitly call out next steps in terms of impact, what's the game plan? How will we hit snapshot 0?

**Pull request merged**